src/Simplify.cpp

```diff
@@ -4585,7 +4589,19 @@ class Simplify : public IRMutator {
             } else {
                 expr = op;
             }
-
+        } else if (op->is_intrinsic(Call::require)) {
+            Expr cond = mutate(op->args[0]);
+            if (can_prove(cond)) {
```

Comment:
This is a bit weird, because can_prove recursively calls a fresh instance of the simplifier, dropping all potentially useful state (e.g. constants to substitute in). Might be better to handle Call::likely explicitly.

Reply:
PTAL